dot/network/service_test.go

```diff
@@ -294,26 +295,81 @@ func TestBroadcastDuplicateMessage(t *testing.T) {
 		Digest: types.NewDigest(),
 	}
 
-	delete(handler.messages, nodeA.host.id())
+	// disable message cache before sending the messages
+	nodeA.host.messageCache = nil
 
-	// Only one message will be sent.
+	// All 5 message will be sent since cache is disabled.
 	for i := 0; i < 5; i++ {
 		nodeA.GossipMessage(announceMessage)
 		time.Sleep(time.Millisecond * 10)
 	}
 
-	time.Sleep(time.Millisecond * 500)
-	require.Equal(t, 2, len(handler.messages[nodeA.host.id()]))
+	messages, _ := handler.messagesFrom(nodeA.host.id())
+	require.Len(t, messages, 6)
```

Comment:
Same question here, can we assert `messages` entirely with `assert.Equal`? Can we predict the messages received?